cmd/agent/subcommands/jmx/command.go

```diff
@@ -166,6 +167,7 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			fx.Provide(func() status.Component { return nil }),
 			fx.Provide(func() eventplatformreceiver.Component { return nil }),
 			fx.Provide(func() optional.Option[collector.Component] { return optional.NewNoneOption[collector.Component]() }),
+			fx.Provide(func() optional.Option[logsAgent.Component] { return optional.NewNoneOption[logsAgent.Component]() }),
```

Comment:
When providing a `NoneOption`, it's simpler to simply use `Supply`.

```suggestion
			fx.Supply(optional.NewNoneOption[logsAgent.Component]()),
```

You could also make that change one line up for the `collector.Component`, while you're here.

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/25234/commits/c39ef1489161ad703d095f1b1eb7fda8a301bb7e